python/ray/data/read_api.py

```diff
@@ -85,6 +86,11 @@
 from ray.util.placement_group import PlacementGroup
 from ray.util.scheduling_strategies import NodeAffinitySchedulingStrategy
 
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
```

Comment:
Not sure if this works with CI yet, but we're not supporting Python 3.8 for Ray 2.8+, so we might not need to guard this

Reply:
Yeah, sounds good to me